### PR TITLE
New: Show banner when doc has xrefs

### PR DESCRIPTION
**Pull request by ConradJChan**
Todo:
- [x] unit tests

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @ConradJChan has signed the CLA. Thanks for the pull request!

File: src/lib/metadataAPI.js [hunk 1]
```diff
@@ -14,17 +17,25 @@ const metadataAPI = {
...
-            }
-
-            throw err;
-        });
+        return metadataAPI
+            .getMetadata(id, SCOPE_GLOBAL, template, options)
+            .then((response) => {
+                // The hasxrefs value is returned as a string 'false' or 'true' so we want
+                // to convert this to a boolean
+                const { [FIELD_HASXREFS]: hasxrefsValue } = response;
```
Comment (jstoffan) [inline src/lib/metadataAPI.js]: Should this be `hasXrefsValue`?

File: src/i18n/en-US.properties [hunk 2]
```diff
@@ -36,6 +36,8 @@ download_file=Download File
 text_truncated=This file has been truncated due to size limits. Please download to view the whole file.
 # Button tooltip to toggle Thumbnails Sidebar
 toggle_thumbnails=Toggle thumbnails
+# Message when file has inaccessible xrefs
+hasxrefs=This preview has links you cannot view, open in its native application to view
```
Comment (jeremypress) [inline src/i18n/en-US.properties]: has_x_refs?

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 3]
```diff
@@ -334,6 +339,23 @@ class DocBaseViewer extends BaseViewer {
...
     }
 
+    /**
+     * Checks for xrefs, depending on file extension
+     * @return {void}
+     */
+    checkForXrefs() {
+        const { extension, id } = this.options.file;
+
+        if (extension === 'dwg') {
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocBaseViewer.js]: Should this live in its own viewer that extends `DocBaseViewer`? I think we generally try to allow the loader to do extension-specific stuff.

File: src/lib/metadataAPI.js [hunk 4]
```diff
@@ -14,17 +17,25 @@ const metadataAPI = {
...
+                return { ...response, [FIELD_HASXREFS]: hasxrefsValue === 'true' };
+            })
+            .catch((err) => {
```
Comment (jeremypress) [inline src/lib/metadataAPI.js]: do we need this? We we're not doing anything with our error anyways.

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 5]
```diff
@@ -334,6 +339,23 @@ class DocBaseViewer extends BaseViewer {
...
+            metadataAPI.getXrefsMetadata(id, TEMPLATE_AUTOCAD, this.options).then((response) => {
+                const { [FIELD_HASXREFS]: hasxrefsValue } = response;
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocBaseViewer.js]: Can be destructured in the function definition.

File: src/lib/metadataAPI.js (hunk 4 above)
Comment (ConradJChan) [reply src/lib/metadataAPI.js]: 🤔 yeah you're right, even if it failed with 400 or any other error code we would just not show a banner. I'll remove the code and related unit tests

File: src/lib/metadataAPI.js [hunk 6]
```diff
@@ -14,16 +17,11 @@ const metadataAPI = {
...
-            // a constructed hasxrefs value
-            if (response && response.status === 404) {
-                return Promise.resolve({ hasxrefs: 'false' });
-            }
-
-            throw err;
+        return metadataAPI.getMetadata(id, SCOPE_GLOBAL, template, options).then((response) => {
+            // The hasxrefs value is returned as a string 'false' or 'true' so we want
+            // to convert this to a boolean
+            const { [FIELD_HASXREFS]: hasXrefsValue } = response;
```
Comment (jstoffan) [inline src/lib/metadataAPI.js]: This could be moved to the function declaration.

File: src/lib/viewers/doc/DocLoader.js [hunk 7]
```diff
@@ -6,6 +6,7 @@ import SinglePageViewer from './SinglePageViewer';
 import RepStatus from '../../RepStatus';
 import { ORIGINAL_REP_NAME, STATUS_SUCCESS } from '../../constants';
 import { DOCUMENT_EXTENSIONS } from '../../extensions';
+import AutoCADViewer from './AutoCADViewer';
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocLoader.js]: Can this live up with the other viewers?

File: src/lib/viewers/doc/DocLoader.js [hunk 8]
```diff
@@ -17,6 +18,12 @@ const VIEWERS = [
         REP: 'pdf',
         EXT: ['gslide', 'gslides', 'odp', 'ppt', 'pptx', 'key']
     },
+    {
+        NAME: 'AutoCAD',
+        CONSTRUCTOR: AutoCADViewer,
+        REP: 'pdf',
+        EXT: ['dwg']
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocLoader.js]: Just to confirm: Is this the only AutoCAD extension we're supporting?

Comment (ConradJChan) [reply src/lib/viewers/doc/DocLoader.js]: Yeah, there was talk of supporting .dwgzip and .dfx but the latest is that this is the only one

**Pull request merged**